analyzers/src/SonarAnalyzer.CSharp/Rules/AsyncVoidMethod.cs

```diff
@@ -77,22 +75,23 @@ public sealed class AsyncVoidMethod : SonarDiagnosticAnalyzer
             || HasAnyMsTestV1AllowedAttribute(methodSymbol);
 
         private static bool IsUsedAsEventHandler(MethodDeclarationSyntax methodDeclaration) =>
-            methodDeclaration.FirstAncestorOrSelf<ClassDeclarationSyntax>() is ClassDeclarationSyntax parentClass
-            && parentClass.DescendantNodes()
+            methodDeclaration.FirstAncestorOrSelf<TypeDeclarationSyntax>(x =>
+                x.IsAnyKind(SyntaxKind.ClassDeclaration, SyntaxKindEx.RecordDeclaration, SyntaxKind.InterfaceDeclaration)) is { } parentDeclaration
```

Comment:
Thanks to the `SyntaxKind.InterfaceDeclaration` here we now support the following case:

 ```
 public interface ISomeInterface
      {
        event EventHandler<bool> MyEvent;

        public void SomeMethod()
        {
            MyEvent += EventHandlerCases_MyEvent;
        }

        private async void EventHandlerCases_MyEvent(object sender, bool e)
        {
            await Task.Run(() => Console.WriteLine("test"));
        }
    }
```
And it would be nice to also cover it in the UTs.

Reply:
Nice one, thank you!